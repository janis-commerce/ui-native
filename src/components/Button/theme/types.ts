export interface SelectedColor {
    light: string,
    main: string,
    dark: string,
}