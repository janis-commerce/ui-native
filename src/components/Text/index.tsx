import React, {ReactElement} from 'react';
import {StyleSheet, Text as TextComponent} from 'react-native';

type TextProps = {
	children?: ReactElement | string;
};

const Text = ({children, ...props}: TextProps) => {
	if (!children) {
		return null;
	}

	return (
		<TextComponent style={styles.TextStyles} {...props}>
			{children}
		</TextComponent>
	);
};

const styles = StyleSheet.create({
	TextStyles: {
		fontSize: 16,
		fontFamily: 'Roboto',
	},
});

export default Text;
