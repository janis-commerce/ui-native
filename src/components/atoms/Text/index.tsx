import React, {ReactElement} from 'react';
import {
	StyleProp,
	StyleSheet,
	Text as TextComponent,
	TextProps as TextComponentProps,
	TextStyle,
} from 'react-native';
import {moderateScale, scaledForDevice} from 'scale';
import {isDevEnv} from 'utils/index';

export interface TextProps extends TextComponentProps {
	children?: ReactElement | string;
	style?: StyleProp<TextStyle>;
}

const Text = ({children, style, ...props}: TextProps) => {
	if (!children) {
		return null;
	}

	const validFontSize = scaledForDevice(13, moderateScale);

	const styles = StyleSheet.create({
		TextStyles: {
			fontSize: validFontSize,
			fontFamily: 'Roboto',
		},
	});

	// istanbul ignore next
	if (isDevEnv()) {
		console.warn('This component is going to be deprecated soon.');
	}
	return (
		<TextComponent style={[styles.TextStyles, style]} {...props}>
			{children}
		</TextComponent>
	);
};

export default Text;
