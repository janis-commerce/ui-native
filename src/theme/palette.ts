import {
	Alert,
	Base,
	Black,
	Env,
	Error,
	GreyScale,
	Primary,
	Success,
	Warning,
	White,
} from '../ts/interfaces/colors';

const primary: Primary = {
	main: '#2979FF',
	dark: '#1759FF',
	light: '#02BFFB',
};
const black: Black = {
	main: '#2F2F2F',
	dark: '#050505',
};
const white: White = {
	main: '#E8EAF6',
	dark: '#D0D3E3',
	light: '#F4F5FB',
};
const grey: GreyScale = {
	100: '#DDDFE2',
	200: '#D5D7DB',
	300: '#C4C6CC',
	400: '#A8AAAC',
	500: '#939598',
	600: '#747679',
	700: '#585858',
};
const base: Base = {
	white: '#fff',
	black: '#000',
};
const success: Success = {
	main: '#1DB779',
	dark: '#109D59',
};
const error: Error = {
	main: '#FF4343',
	dark: '#FF2A2A',
};
const warning: Warning = {
	main: '#FF8D10',
	dark: '#FF6E08',
};
const alert: Alert = {
	main: '#FFCE17',
	dark: '#FFBA0C',
};
const environment: Env = {
	qa: '#1DB779',
	beta: '#F13B70',
};

export {primary, black, white, grey, base, success, error, warning, alert, environment};
