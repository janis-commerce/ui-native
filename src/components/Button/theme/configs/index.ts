import {moderateScale, scaledForDevice} from '../../../../scale';
import {base, black, grey, white} from '../../../../theme/palette';
import {SelectedColor} from '../../types';

export const colorConfig = (selectedColor: SelectedColor) => {
	return {
		main: {
			background: {
				contained: selectedColor.main,
				outlined: selectedColor.light,
				text: 'transparent',
			},
			border: {
				contained: 'transparent',
				outlined: grey[300],
				text: 'transparent',
			},
			text: {
				main: {
					contained: base.white,
					outlined: black.main,
					text: black.main,
				},
				secondary: {
					contained: base.white,
					outlined: selectedColor.main,
					text: selectedColor.main,
				},
			},
		},
		pressed: {
			background: {
				contained: selectedColor.dark,
				outlined: selectedColor.light,
				text: white.main,
			},
			border: {
				contained: 'transparent',
				outlined: selectedColor.main,
				text: 'transparent',
			},
			text: {
				main: {
					contained: base.white,
					outlined: black.main,
					text: black.main,
				},
				secondary: {
					contained: base.white,
					outlined: selectedColor.main,
					text: selectedColor.dark,
				},
			},
		},
		disabled: {
			background: {
				contained: grey[200],
				outlined: grey[100],
				text: 'transparent',
			},
			border: {
				main: {
					contained: 'transparent',
					outlined: 'transparent',
					text: 'transparent',
				},
				secondary: {
					contained: 'transparent',
					outlined: grey[300],
					text: 'transparent',
				},
			},
			text: {
				main: {
					contained: grey[400],
					outlined: grey[300],
					text: grey[300],
				},
				secondary: {
					contained: grey[400],
					outlined: grey[300],
					text: grey[300],
				},
			},
		},
	};
};

export const stlyeConfig = {
	container: {
		height: {
			main: scaledForDevice(50, moderateScale),
			secondary: scaledForDevice(42, moderateScale),
		},
		shadow: {
			// android
			shadowColor: '#000',
			elevation: 5,
			// IOS
			shadowOpacity: 0.25,
			shadowRadius: 3.84,
			shadowOffset: {
				width: 0,
				height: scaledForDevice(3, moderateScale),
			},
		},
	},
	directionWrapper: {
		center: {
			justifyContent: 'center',
			alignItems: 'center',
		},
		flexDirection: {
			left: 'row',
			right: 'row-reverse',
			top: 'column',
			bottom: 'column-reverse',
		},
	},
	text: {
		fontWeight: '500',
		textAlign: 'center',
	},
	icon: {
		margin: {
			top: {},
			bottom: {},
			left: {
				marginRight: scaledForDevice(8, moderateScale),
			},
			right: {
				marginLeft: scaledForDevice(8, moderateScale),
			},
		},
	},
};
