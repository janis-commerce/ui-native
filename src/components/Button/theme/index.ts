import { alert, base, black, error, grey, primary, success, warning} from '../../../theme/palette';
import type { SelectedColor } from './types';

export const themeColors = {
    primary: {
        light: base.white,
        main: primary.main,
        dark: primary.dark,
    },
    secondary: {
        light: base.white,
        main: black.main,
        dark: black.dark,
    },
    success: {
        light: base.white,
        main: success.main,
        dark: success.dark,
    },
    error: {
        light: base.white,
        main: error.main,
        dark: error.dark,
    },
    warning: {
        light: base.white,
        main: warning.main,
        dark: warning.dark,
    },
    alert: {
        light: base.white,
        main: alert.main,
        dark: alert.dark,
    }
}

export const mixedColors = (selectedColor: SelectedColor) => {
    return {
        mainBgColor: {
            contained:  selectedColor.main,
            outlined: selectedColor.light,
            text: 'transparent'
        },
        mainBorderColor: {
            contained:  'transparent',
            outlined: grey[300],
            text: 'transparent'
        },
        pressedBgColor: {
            contained:  selectedColor.dark,
            outlined: selectedColor.light,
            text: 'transparent'
        },
        pressedBorderColor: {
            contained:  'transparent',
            outlined: selectedColor.main,
            text: 'transparent'
        },
        disabledBgColor: {
            contained:  grey[200],
            outlined: grey[100],
            text: 'transparent'
        },
        disabledBorderColor: {
            main: {
                contained: 'transparent',
                outlined: 'transparent',
                text: 'transparent'
            },
            secondary: {
                contained: 'transparent',
                outlined: grey[300],
                text: 'transparent'
            }
        }
    }
}