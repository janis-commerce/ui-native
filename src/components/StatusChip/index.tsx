import React, {ReactElement, isValidElement} from 'react';
import {StyleSheet, Text, ViewProps, View} from 'react-native';
import {base, grey, primary} from '../../theme/palette';

interface StatusChipProps extends ViewProps {
	children?: ReactElement | string;
	background?: string;
}

const StatusChip = ({children, ...props}: StatusChipProps) => {
	const isChildrenAString = typeof children === 'string';
	const isCustomComponent = isValidElement(children);

	const hasToRenderStatusChip = !children || (!isChildrenAString && !isCustomComponent);

	if (hasToRenderStatusChip) {
		return null;
	}

	return (
		<View style={styles(props).ViewStyles} {...props}>
			{isCustomComponent ? children : <Text style={styles(props).TextStyles}>{children}</Text>}
		</View>
	);
};

const styles = ({background}: StatusChipProps) =>
	StyleSheet.create({
		ViewStyles: {
			height: 24,
			flexDirection: 'row',
			alignItems: 'center',
			paddingLeft: 12,
			paddingRight: 12,
			borderRadius: 12,
			backgroundColor: background ?? base.white,
			borderWidth: 1,
			borderColor: background ?? grey['300'],
		},
		TextStyles: {
			fontSize: 15,
			lineHeight: 18,
			fontFamily: 'Roboto',
			fontWeight: '900',
			textAlign: 'center',
			color: background ? base.white : primary.main,
		},
	});

export default StatusChip;
