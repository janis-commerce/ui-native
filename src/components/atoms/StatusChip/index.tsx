import React, {ReactElement, isValidElement} from 'react';
import {StyleSheet, ViewProps, View} from 'react-native';
import {base, grey, primary} from 'theme/palette';
import Text from 'atoms/Text';
import {horizontalScale, moderateScale, scaledForDevice} from 'scale';

interface StatusChipProps extends ViewProps {
	children?: ReactElement | string;
	background?: string;
}

const validHeight = scaledForDevice(24, moderateScale);
const validPadding = scaledForDevice(12, horizontalScale);
const validBorderRadius = scaledForDevice(12, moderateScale);
const validBorderWidth = scaledForDevice(1, moderateScale);
const validFontSize = scaledForDevice(13, moderateScale);
const validLineHeight = scaledForDevice(18, moderateScale);

const styles = ({background}: StatusChipProps) =>
	StyleSheet.create({
		ViewStyles: {
			height: validHeight,
			flexDirection: 'row',
			alignItems: 'center',
			paddingLeft: validPadding,
			paddingRight: validPadding,
			borderRadius: validBorderRadius,
			backgroundColor: background ?? base.white,
			borderWidth: validBorderWidth,
			borderColor: background ?? grey['300'],
		},
		TextStyles: {
			fontSize: validFontSize,
			lineHeight: validLineHeight,
			fontFamily: 'Roboto',
			fontWeight: '900',
			textAlign: 'center',
			color: background ? base.white : primary.main,
		},
	});

const StatusChip = ({children, ...props}: StatusChipProps) => {
	const isChildrenAString = typeof children === 'string';
	const isCustomComponent = isValidElement(children);

	const hasToRenderStatusChip = !children || (!isChildrenAString && !isCustomComponent);

	if (hasToRenderStatusChip) {
		return null;
	}

	return (
		<View style={styles(props).ViewStyles} {...props}>
			{isCustomComponent ? children : <Text style={styles(props).TextStyles}>{children}</Text>}
		</View>
	);
};

export default StatusChip;
