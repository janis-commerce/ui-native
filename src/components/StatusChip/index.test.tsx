import 'react-native';
import React from 'react';
import {create} from 'react-test-renderer';
import StatusChip from './index';
import Text from '../Text';
import {primary} from '../../theme/palette';

describe('StatusChip component', () => {
	describe('return error because is not a valid children', () => {
		it('undefined children', () => {
			const {toJSON} = create(<StatusChip />);
			expect(toJSON()).toBeNull();
		});
	});

	describe('render correct', () => {
		it('text is passed', () => {
			const {toJSON} = create(<StatusChip>Partially delivered</StatusChip>);
			expect(toJSON()).toMatchSnapshot();
		});

		it('if a color is passed', () => {
			const {toJSON} = create(<StatusChip color={primary.main}>Delivered</StatusChip>);
			expect(toJSON()).toMatchSnapshot();
		});

		it('a valid component is passed', () => {
			const {toJSON} = create(
				<StatusChip>
					<Text>Custom component</Text>
				</StatusChip>
			);
			expect(toJSON()).toMatchSnapshot();
		});
	});
});
