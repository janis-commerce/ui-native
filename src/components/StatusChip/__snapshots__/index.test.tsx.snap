// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusChip component render correct a valid component is passed 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#C4C6CC",
      "borderRadius": 12,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 24,
      "paddingLeft": 12,
      "paddingRight": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "fontSize": 13,
        },
        undefined,
      ]
    }
  >
    Custom component
  </Text>
</View>
`;

exports[`StatusChip component render correct if a background is passed 1`] = `
<View
  background="#2979FF"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2979FF",
      "borderColor": "#2979FF",
      "borderRadius": 12,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 24,
      "paddingLeft": 12,
      "paddingRight": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "fontSize": 13,
        },
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 13,
          "fontWeight": "900",
          "lineHeight": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    Delivered
  </Text>
</View>
`;

exports[`StatusChip component render correct text is passed 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#C4C6CC",
      "borderRadius": 12,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 24,
      "paddingLeft": 12,
      "paddingRight": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "fontSize": 13,
        },
        Object {
          "color": "#2979FF",
          "fontFamily": "Roboto",
          "fontSize": 13,
          "fontWeight": "900",
          "lineHeight": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    Partially delivered
  </Text>
</View>
`;
